### ISSUE TITLE
nexmark q5 long running OOM due to unawareness of container memory limit

### ISSUE THREAD

**Issue by KeXiangWang** [risingwavelabs/risingwave#6615]
### Describe the bug

When running nexmark sql5, risingwave ComputeNode will OOM after about 30 mins.

### To Reproduce

the bug emerges in EKS environment, but should also happen in local environment.
use nexmark-bench to generate data through Kafka.

### Additional context

Based on metrics, the OOM happens on both of the two hashagg fragment(max and count). 

Here's the nexmark q5
```
    CREATE MATERIALIZED VIEW nexmark_q5
    AS
    SELECT AuctionBids.auction,
           AuctionBids.num
    FROM (SELECT bid.auction,
                 count(*)     AS num,
                 window_start AS starttime
          FROM
              HOP(bid, date_time, INTERVAL '2' SECOND, INTERVAL '10' SECOND)
          GROUP BY window_start,
                   bid.auction) AS AuctionBids
    JOIN (SELECT max(CountBids.num) AS maxn,
                CountBids.starttime_c
          FROM (SELECT count(*)     AS num,
                      window_start AS starttime_c
                FROM HOP(bid, date_time, INTERVAL '2' SECOND, INTERVAL '10' SECOND)
                GROUP BY bid.auction,
                        window_start) AS CountBids
          GROUP BY CountBids.starttime_c) AS MaxBids
    ON
                AuctionBids.starttime = MaxBids.starttime_c AND
                AuctionBids.num >= MaxBids.maxn;
```

**Comment by fuyufjh** [comment]
Hi, does someone have ideas about the reason?

**Comment by BugenZhao** [comment]
> Hi, does someone have ideas about the reason?

I'll help to investigate it. 👀

**Comment by KeXiangWang** [comment]
One possible reason. 
Here's the [codes](https://github.com/risingwavelabs/risingwave/blob/3474214f9d604aa908a8e8c6454ca3aec27dedde/src/common/src/config.rs#L400) we get the memory size:
```
        let mut sys = System::new();
        sys.refresh_memory();
        sys.total_memory() as usize
```
In EKS environment, the compute node is encapsulated in K8S pod. One VM(EC2) can have multiple pod. The resource is limited at pod level. But the API fetchs the VM system information. 
For example, if we have one VM's memory size is 16G and 4 pods on it. And the pod resource limitation is 4G. 
The above codes get 16G.
The our lru is evicted based on the memory size.

**Comment by KeXiangWang** [comment]
https://github.com/risingwavelabs/risingwave/issues/6536
Same Issue

**Comment by BugenZhao** [comment]
Is this the same as https://github.com/risingwavelabs/risingwave/issues/6536?

**Comment by lmatz** [comment]
Related: #6536 

Some ideas:
1. LRU manager should strictly keep memory under 4G 
2. Cloud configures `all_memory_available_bytes` correctly, i.e. 4G in this case.
3. But ideally, kernel can also do a check (being container aware) whether the configured parameter makes sense or not, and log out a ERROR/WARN message.

**Comment by lmatz** [comment]
This should be fixed by #6536, close it now.
